src/js/webrequest.js

```diff
@@ -172,6 +172,23 @@ function onBeforeSendHeaders(details) {
     } else {
       return {};
     }
+
+  } else if (badger.isSpoofReferrerEnabled()) {
```

Comment:
This shouldn't be an "else if", just an "if", since the "if this isn't a third-party request" block above always exits.

Reply:
We should move this entire referrer spoofing block down in `onBeforeSendHeaders` callback.

We don't want to do this when Badger is disabled, and we already have a check for this later on in the callback (`if (!badger.isPrivacyBadgerEnabled(tabDomain)) ...`). We don't want to do this when Badger ends up blocking the request (because it happens to be a third strike with cookie headers).

We do want this to work in concert with DNT header injection though, so it makes sense for things that alter headers to live next to each other.